Traceback (most recent call last):
  File "C:\Users\ckeulen\anaconda3\Lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\Users\ckeulen\anaconda3\Lib\site-packages\nbclient\client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ckeulen\anaconda3\Lib\site-packages\jupyter_core\utils\__init__.py", line 173, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ckeulen\anaconda3\Lib\asyncio\base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "C:\Users\ckeulen\anaconda3\Lib\site-packages\nbclient\client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "C:\Users\ckeulen\anaconda3\Lib\site-packages\nbclient\client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\ckeulen\anaconda3\Lib\site-packages\nbclient\client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# import the libraries we'll need
import CoolProp.CoolProp as CP
import numpy as np

#define variables
fluid = 'water'
R = 0.4615   #hydrogen gas constant kJ/kg.K
C_p = 1.8723   #hydrogen Cp kJ/kg.K

P_1 = 101e+3   #initial pressure in Pa
T_1 = 110 + 273.15   #initial temperature in K
T_2 = 120 + 273.15   #final temperature in K

D_1 = CP.PropsSI("D", "T", T_1, "P", P_1, fluid)   #initial density in kg/m3
D_2 = D_1   #constant density

P_2 = CP.PropsSI("P", "T", T_2, "D", D_2, fluid)   #final pressure based on temperature and density in Pa

ds_a = C_p * np.log(T_2/T_1) - R * np.log(P_2/P_1)   #changes in entropy in kJ/kg.k

print('The entropy change using ideal gas assumption is:', f"{ds_a:.3f}", 'kJ/kg.K')
------------------


[1;31m---------------------------------------------------------------------------[0m
[1;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [1;32mIn[2], line 2[0m
[0;32m      1[0m [38;5;66;03m# import the libraries we'll need[39;00m
[1;32m----> 2[0m [38;5;28;01mimport[39;00m [38;5;21;01mCoolProp[39;00m[38;5;21;01m.[39;00m[38;5;21;01mCoolProp[39;00m [38;5;28;01mas[39;00m [38;5;21;01mCP[39;00m
[0;32m      3[0m [38;5;28;01mimport[39;00m [38;5;21;01mnumpy[39;00m [38;5;28;01mas[39;00m [38;5;21;01mnp[39;00m
[0;32m      5[0m [38;5;66;03m#define variables[39;00m

[1;31mModuleNotFoundError[0m: No module named 'CoolProp'

